src/Howerpoint.hs

```diff
@@ -59,34 +83,39 @@ help = mapM_ (putStrLn . fst . colorize)
 
 
 data Presentation = Presentation {
-  n :: IO (),
-  p :: IO (),
-  nn :: Int -> IO (),
-  pp :: Int -> IO (),
-  g :: Int -> IO (),
-  resetSize :: IO (),
-  setSize :: Int -> Int -> IO (),
-  showSize :: IO (),
+  n             :: IO (),
+  p             :: IO (),
+  nn            :: Int -> IO (),
+  pp            :: Int -> IO (),
+  g             :: Int -> IO (),
+  resetSize     :: IO (),
+  setSize       :: Int -> Int -> IO (),
+  showSize      :: IO (),
   codeFromSlide :: IO [String]
 }
 
-parseSlides :: String -> IO [Slide]
+parseSlides :: String -> IO (Zipper Slide)
 parseSlides path = do
     h <- openFile path ReadMode
-    slides <- (splitOn ["---"]) . lines <$> hGetContents h
-    slides `deepseq` hClose h
-    pure slides
+    slides' <- splitOn ["---"] . lines <$> hGetContents h
+    slides' `deepseq` hClose h
+    return $ fromList slides'
 
 displaySlide :: IORef PresentationState -> IO ()
 displaySlide ps = do
-  ps <- readIORef ps
-  let slide = getCurrentSlide ps
-  mapM putStrLn (formatSlide ps slide) >> pure ()
-
-moveSlide :: IORef PresentationState -> (Int -> Int) -> IO ()
-moveSlide ps f = do
-  modifyIORef ps (\p -> p {currentSlide = f (currentSlide p)})
-  displaySlide ps
+  ps' <- readIORef ps
+  mapM_ putStrLn (formatSlide ps' (getCurrentSlide ps'))
+
+moveSlide :: (Int -> Int) -> PresentationState -> PresentationState
+moveSlide f ps
+  = maybe ps (\ss -> ps {slides = ss, currentSlide = newpos}) slides'
```

Comment:
We try to move to the position given by applying f to the current position, if we succeed (that is we have enough slides), we go there, otherwise stay where we are.

This means that jumping to from 0 to 10 when we have only 7 slides will stay at 0, and not move to 7. This can be easily changed however, but I felt like this is a more intuitive behaviour